src/v/archival/segment_reupload.cc

```diff
@@ -147,7 +172,9 @@ void segment_collector::do_collect(segment_collector_mode mode) {
 
         // For the first segment found, begin offset needs to be
         // re-aligned if it falls inside manifest segment.
-        if (_segments.empty()) {
+        if (
+          _segments.empty()
+          && mode == segment_collector_mode::collect_compacted) {
```

Comment:
Would you need to align the begin offset to manifest for the adjacent segment collection?

Reply:
Nope, they should always be aligned to the manifest already, since the offset range for adjacent segment collection comes from the manifest itself.